### PR TITLE
Tweak the normalization function for chromosome name

### PR DESCRIPTION
**Pull request by federkasten**
This PR adds following features,

- Parse a suffix which represents contig type and version (e.g. `v2_alt`)
- Keep that suffix in lower case to work with reference files that are available from UCSC
- Add a function to determine whether chromosome name is a primary one

### CODE REVIEW

File: src/cljam/util/chromosome.clj [hunk 1]
```diff
@@ -13,6 +13,11 @@
   [s]
   (cstr/replace s #"(?i)^chr" ""))
 
+(defn- split-version-suffix
+  [s]
+  (let [[_ base suffix leftover] (re-matches #"(.+?)((?i)v[0-9](_alt|_random)?)?$" s)]
```
Comment (totakke) [inline src/cljam/util/chromosome.clj]: `leftover`  is never used, please replace it with `_`. `$` is not needed for `re-matches`.

File: src/cljam/util/chromosome.clj [hunk 2]
```diff
@@ -32,7 +37,14 @@
...
+  (let [[base version-suffix] (split-version-suffix s)]
+    (str (-> base
+             normalize-name
+             prepend-chromosome-prefix
+             normalize-chromosome-prefix)
+         (if version-suffix (cstr/lower-case version-suffix)))))
+
+(defn is-primary-chromosome?
+  [s]
+  (not (nil? (re-matches #"^chr([0-9]{1,2}|X|Y|M|MT)$"
```
Comment (totakke) [inline src/cljam/util/chromosome.clj]: `not nil?` can be replaced with `some?`, and `^$` is not needed for `re-matches`.

**Comment by federkasten** [comment]
Thanks for reviewing! I have fixed the points.

**Comment by totakke** [comment]
Thank you!

**Pull request merged**